### DISCUSSION TITLE
Future of the Image class

### DISCUSSION THREAD

**Discussion by ghost** [emgucv/emgucv#670] [Q&A]
I am just getting into working with the EmguCV library and I am confused about the future of the Emgu.CV.Image class. I was enjoying the chain-able nature (among other things) of the Image class and was considering if it was possible to abandon the CvInvoke calls altogether. I was directed to the [Working with images](https://www.emgu.com/wiki/index.php/Working_with_Images) page on the Wiki and read that the Image class is being phased out since version 3. This is the only mention of the supposed deprecation of the class that I have been able to find and I didn't notice anything about the phase out in the API documentation. Is this still the case or will the Image class make a comeback? Is it prudent to avoid the Image class altogether?

**Comment by emgucv** [comment]
It depends on when Open CV will completely remove the "IplImage*"  structure, AFAIK, they are working on it. Once it is removed from Open CV, the Image<,> class will be removed from Emgu CV. 

Open CV deprecated IplImage* since v3, it is quite likely it will be removed in v5.  We are also moving our codes away from the use of the Image<,> class towards Mat.
